### PR TITLE
librbd: group and snapshot cleanup

### PR DESCRIPTION
**Pull request by dillaman**
Enable the operations feature w/ group operation when an image is added to a group. Cleanup and normalize the API and CLI for groups. Add a simple method for retrieving snapshots from the OSDs.

### CODE REVIEW

File: src/test/cli/rbd/help.t [hunk 1]
```diff
@@ -9,107 +9,109 @@ Skip test on FreeBSD as it generates different output there.
...
-      watch                             Watch events on image.
+      bench                               Simple benchmark.
+      children                            Display children of snapshot.
+      clone                               Clone a snapshot into a COW child image.
+      copy (cp)                           Copy src image to dest.
+      create                              Create an empty image.
+      diff                                Print extents that differ since a
+                                          previous snap, or image creation.
+      disk-usage (du)                     Show disk usage stats for pool, image
+                                          or snapshot
```
Comment (runsisi) [inline src/test/cli/rbd/help.t]: need a period(`.`) at the end of the help message

Comment (dillaman) [reply src/test/cli/rbd/help.t]: (unrelated to this PR)

File: src/pybind/rbd/rbd.pyx [hunk 2]
```diff
@@ -2062,6 +2094,53 @@ cdef class Image(object):
...
+        if ret != 0:
+            raise make_ex(ret, 'error getting group for image %s' % (self.name,))
+        result = {
+            'pool' : info.pool,
+            'name' : decode_cstr(info.name)
+            }
+        rbd_group_info_cleanup(&info, sizeof(info))
+        return result
+
+    def id(self):
```
Comment (trociny) [inline src/pybind/rbd/rbd.pyx]: @dillaman I suppose this redefinition of `id()` method was added unintentionally here?

File: src/pybind/rbd/rbd.pyx [hunk 3]
```diff
@@ -2062,6 +2094,53 @@ cdef class Image(object):
...
 
+    def group(self):
+        """
+        Get information about the image's group.
+
+        :returns: dict - contains the following keys:
+
+            * ``pool`` (int) - id of the group pool
+
+            * ``name`` (ste) - name of the group
```
Comment (trociny) [inline src/pybind/rbd/rbd.pyx]: typo: s/ste/str/

File: src/pybind/rbd/rbd.pyx (hunk 2 above)
Comment (dillaman) [reply src/pybind/rbd/rbd.pyx]: Odd -- unintended indeed.

**Pull request merged**